wgengine/magicsock/magicsock.go

```diff
@@ -124,6 +124,8 @@ type Options struct {
 	// EndpointsFunc optionally provides a func to be called when
 	// endpoints change. The called func does not own the slice.
 	EndpointsFunc func(endpoint []string)
+
+	derpTLSConfig *tls.Config // normally nil; used by tests
```

Comment:
We'll probably want to export this at some point, but we can do that later.

Reply:
I figured minimizing the exposed surface is preferable for now (but also not having tests reach into the Conn's private fields). I liked the convention of "private options for tests". But I agree, we'll likely want to poke at this from other tests.